### DISCUSSION TITLE
Manager says that LSPosed is not installed

### DISCUSSION THREAD

**Discussion by LucasBS1** [LSPosed/LSPosed#1988] [Q&A]
Downloaded LSPosed from github and installed it normaly on Magisk.
But there's no prompt for a shortcut, so I install the manager from the folder on the description 
(I had to move the APK away from that folder first. The installation fails if attempted there)

The manager installs, but says LSPosed is not installed

What is going on ?
I suspect Magisk is somehow to blame, despite everything else working

Android 9
Magisk 24.3
Tried from V1.7.0 to the latest V1.8.3-6552

Also, typing those numbers on the dial doesn't bring anything.

**Comment by yujincheng08** [comment]
how can we know what is going on in your device? ship your device to us?

**Reply by LucasBS1** [reply]
.

_Originally posted by @yujincheng08 in https://github.com/LSPosed/LSPosed/issues/1981#issuecomment-1153165985_

>> Why is this thing closed if the problem remains ?
>
>because its fixed

It's not

I could give a screenshot, but the color would be the only difference between mine's and the screenshots from yunyu950908 bug report.

Thank you to the developers and those willing to help, by the way

**Comment by LucasBS1** [comment]
_Originally posted by @Stillhard in https://github.com/LSPosed/LSPosed/issues/1981#issuecomment-1153166591_


>Try [this artifact](https://github.com/LSPosed/LSPosed/actions/runs/2466592981#artifacts)

Thank you for the links.
But didn't work...
And I triple checked: LSPosed 1.8.3-6568.

Rebooted many times after uninstalling the previous (6552) and installing the new 

Also made sure that both the framework and the manager are of the same version

**Reply by mywalkb** [reply]
You should post debug logs, not always "LSPosed not installed" has the same cause.
You have android 9 and in that issue android is 12. You have magisk 24.3 and in that issue Magisk is 25.

**Reply by LucasBS1** [reply]
There is no log on the _root/data/adb/lspd/_. There's only the manager APK in that folder. 
_LSPosed-v1.8.3-6568-zygisk-debug_ is installed, with it's corresponding manager. 
The option to enable/disable verbose log is grayed out in the manager settings.

There is a bunch of folders on _root/data/adb/modules/zygisk_lsposed_, but don't suppose these are of interest.

**Reply by mywalkb** [reply]
Try to post the output of logcat, if there is some errors should appear.

> The option to enable/disable verbose log is grayed out in the manager settings.

because the daemon don't start

**Reply by LucasBS1** [reply]
Is this the [logcat ?](https://raw.githubusercontent.com/LucasBS1/Logcats/main/Logcat-12Jun22)

Extracted using **_adb logcat_** >logcat.txt, as soon as I turned on the phone and enabled debugger.

As a curiosity, I passed my eyes on this log and found many references to _LSPosed-v1.8.3-6552-**riru**-release.zip_
I'm certain that _LSPosed-v1.8.3-6568-**zygisk**-debug_  is installed

**Reply by mywalkb** [reply]
> Is this the [logcat ?](https://raw.githubusercontent.com/LucasBS1/Logcats/main/Logcat-12Jun22)
> 
> Extracted using **_adb logcat_** >logcat.txt, as soon as I turned on the phone and enabled debugger.

ok but something is missing

> As a curiosity, I passed my eyes on this log and found many references to _LSPosed-v1.8.3-6552-**riru**-release.zip_ I'm certain that _LSPosed-v1.8.3-6568-**zygisk**-debug_ is installed

This is strange. Have you manager parasitic or installed?
Post the output of command `ls -l /data/adb/modules/
` executed as root

**Reply by mywalkb** [reply]
> Is this the [logcat ?](https://raw.githubusercontent.com/LucasBS1/Logcats/main/Logcat-12Jun22)
> 
> Extracted using **_adb logcat_** >logcat.txt, as soon as I turned on the phone and enabled debugger.

The log is too late, if is taken after boot, you should increase the buffer size in developer settings.

> As a curiosity, I passed my eyes on this log and found many references to _LSPosed-v1.8.3-6552-**riru**-release.zip_ I'm certain that _LSPosed-v1.8.3-6568-**zygisk**-debug_ is installed

Ignore previous reply. This happen when the binder is not alive. The default api is riru, so manager check update for riru when the daemon don't start.
The information if you use parasitic or installed in good to know.

**Reply by LucasBS1** [reply]
Thank you for the second day of support

I had to install the manager because the parasitic doesn't launch by dialing x#x#5776733#x#x, nor did Magisk offer to create a shortcut after the installation of the framework.

- Here's the [result of the ls command](https://raw.githubusercontent.com/LucasBS1/Logcats/6bdf89ba241f5b6eac61347eb580adc367a6639e/LSResult-14Jun22)

- Took the liberty of including the [install log](https://raw.githubusercontent.com/LucasBS1/Logcats/main/LSPosedInstallLog) of LSPosed

- Here's the 16MB buffered [LogCat](https://raw.githubusercontent.com/LucasBS1/Logcats/main/LogCat-14Jun22), taken after boot, as soon as I turn on the Debugger that insists on turning itself off


I disabled a few modules in Magisk and froze some apps in Titanium before taking the logcat in the hopes of clearing some pollution.

**Reply by mywalkb** [reply]
When daemon doesn't start parasitic don't work. Magisk don't ask to you to make a shortcut for LSPosed, the daemon ask to you.
Now log start from the begin, I understand the issue but I can't find the cause for now. 
Have you tried to disable all other modules on magisk except LSPosed?

**Reply by LucasBS1** [reply]
> Have you tried to disable all other modules on magisk except LSPosed?

Attempted now. Uninstalled the others and rebooted prior. No difference, unfortunately.

Forgot to mention: BEFORE all this, I also tried reflashing the whole ROM, wiping everything, rooting from scratch... No difference either.

But thank you for trying. I'm sure a solution will eventually appear.

**Reply by mywalkb** [reply]
I think [this](https://github.com/LSPosed/LSPosed/issues/785) is your issue.

```
06-14 01:29:11.979  1335  1335 I ServiceManagement: Removing namespace from process name android.hardware.biometrics.fingerprint@2.1-fpcservice to fingerprint@2.1-fpcservice.
06-14 01:29:12.003   948   948 W lspd    : Landroid/os/Build; failed initialization: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.lang.String.startsWith(java.lang.String)' on a null object reference
06-14 01:29:12.003  1340  1340 W lspd    : Landroid/os/Build; failed initialization: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.lang.String.startsWith(java.lang.String)' on a null object reference
06-14 01:29:12.003   948   948 W lspd    :   at java.lang.String android.os.Build.getBuildNumber() (Build.java:1207)
06-14 01:29:12.003  1340  1340 W lspd    :   at java.lang.String android.os.Build.getBuildNumber() (Build.java:1207)
06-14 01:29:12.004   948   948 W lspd    :   at java.lang.String android.os.Build.deriveFingerprint() (Build.java:959)
06-14 01:29:12.004  1340  1340 W lspd    :   at java.lang.String android.os.Build.deriveFingerprint() (Build.java:959)
06-14 01:29:12.004   948   948 W lspd    :   at void android.os.Build.<clinit>() (Build.java:945)
06-14 01:29:12.004  1340  1340 W lspd    :   at void android.os.Build.<clinit>() (Build.java:945)
06-14 01:29:12.004   948   948 W lspd    :   at java.lang.String android.os.Build.access$000(java.lang.String) (Build.java:42)
06-14 01:29:12.004  1340  1340 W lspd    :   at java.lang.String android.os.Build.access$000(java.lang.String) (Build.java:42)
06-14 01:29:12.004   948   948 W lspd    :   at void android.os.Build$VERSION.<clinit>() (Build.java:201)
06-14 01:29:12.004  1340  1340 W lspd    :   at void android.os.Build$VERSION.<clinit>() (Build.java:201)
06-14 01:29:12.004   948   948 W lspd    :   at java.lang.String com.android.internal.os.RuntimeInit.getDefaultUserAgent() (RuntimeInit.java:258)
06-14 01:29:12.004  1340  1340 W lspd    :   at java.lang.String com.android.internal.os.RuntimeInit.getDefaultUserAgent() (RuntimeInit.java:258)
06-14 01:29:12.004   948   948 W lspd    :   at void com.android.internal.os.RuntimeInit.commonInit() (RuntimeInit.java:225)
06-14 01:29:12.004  1340  1340 W lspd    :   at void com.android.internal.os.RuntimeInit.commonInit() (RuntimeInit.java:225)
06-14 01:29:12.004   948   948 W lspd    :   at void com.android.internal.os.RuntimeInit.main(java.lang.String[]) (RuntimeInit.java:334)
06-14 01:29:12.004  1340  1340 W lspd    :   at void com.android.internal.os.RuntimeInit.main(java.lang.String[]) (RuntimeInit.java:334)
06-14 01:29:12.004   948   948 W lspd    : 
06-14 01:29:12.004  1340  1340 W lspd    : 
06-14 01:29:12.004   948   948 W lspd    : Landroid/os/Build$VERSION; failed initialization: java.lang.ExceptionInInitializerError: 
06-14 01:29:12.004  1340  1340 W lspd    : Landroid/os/Build$VERSION; failed initialization: java.lang.ExceptionInInitializerError: 
06-14 01:29:12.004  1340  1340 W lspd    :   at java.lang.String android.os.Build.access$000(java.lang.String) (Build.java:42)
06-14 01:29:12.004   948   948 W lspd    :   at java.lang.String android.os.Build.access$000(java.lang.String) (Build.java:42)
06-14 01:29:12.004  1340  1340 W lspd    :   at void android.os.Build$VERSION.<clinit>() (Build.java:201)
06-14 01:29:12.004   948   948 W lspd    :   at void android.os.Build$VERSION.<clinit>() (Build.java:201)
06-14 01:29:12.004  1340  1340 W lspd    :   at java.lang.String com.android.internal.os.RuntimeInit.getDefaultUserAgent() (RuntimeInit.java:258)
06-14 01:29:12.004   948   948 W lspd    :   at java.lang.String com.android.internal.os.RuntimeInit.getDefaultUserAgent() (RuntimeInit.java:258)
06-14 01:29:12.004  1340  1340 W lspd    :   at void com.android.internal.os.RuntimeInit.commonInit() (RuntimeInit.java:225)
06-14 01:29:12.004   948   948 W lspd    :   at void com.android.internal.os.RuntimeInit.commonInit() (RuntimeInit.java:225)
06-14 01:29:12.004  1340  1340 W lspd    :   at void com.android.internal.os.RuntimeInit.main(java.lang.String[]) (RuntimeInit.java:334)
06-14 01:29:12.004   948   948 W lspd    :   at void com.android.internal.os.RuntimeInit.main(java.lang.String[]) (RuntimeInit.java:334)
06-14 01:29:12.004  1340  1340 W lspd    : 
06-14 01:29:12.004   948   948 W lspd    : 
```

Try these solutions: [1](https://github.com/LSPosed/LSPosed/issues/785#issuecomment-869190719) [2](https://github.com/LSPosed/LSPosed/issues/785#issuecomment-869196215)

**Reply by LucasBS1** [reply]
.
HEADSHOT !

Man, you guys are MAGICIANS ! 

I can't believe you discovered something like THAT just by staring at a log !

Anyway.
Since I'm lazy, I didn't follow solutions 1 or 2. Instead, when they mentioned fingerprint, I downloaded a module called MagiskHide Props Config, that did the job for me here. Works on terminal emulator.

As soon as I rebooted, the shortcut prompt popped up. Amazing !

Now, why I have to spoof my own device, god knows...

Edit: switched to solution 1. It works well enough. Otherwise I would be slave of that Props Config module

Thank you for the support and the effective solution

By the way, I spent a day googling for "LSPosed is not installed", but that discussion you linked never showed up. Sad, because it would have made things faster...
Apparently, I wasn't the only one that didn't find that thread, since when I started this one, I linked another recent thread of the same subject...

**Reply by winerider** [reply]
Hey, I'm in a similar sort of situation. My device is OnePlus 9rt running stock rom with Android 13. lsposed worked fine before the last OTA update but after the update when i tried to reinstall it, it just gets installed but there's no shortcut prompt and the manager app says not installed.

I know you've mentioned the solutions already but I can't really understand the technicalities. Can you please explain in a simpler manner ?

**Comment by Silenti0** [comment]
Running android 12 on a Pixel 3 
Latest TWRP Recovery, rooted with Magisk 25.2 
Have tried everything to get LSPosed running 
I have tried flashing different versions of Magisk and running both the Riru and Zygisk routes for Lsposed modules
When they flash, everything looks successful on Magisk and it says LSPosed is loaded in my notificatoin bar but the manager, no matter what, says that lsposed is not installed and reccommends I install it.
I have tried different versions of LsPosed and looked at other forums (like this one) for workarounds but just cant get anything to be recognized by the manager as installed so that none of the xposed framework mods I want to run will work.

Attached are my boot log, magisk logs, and system info.

Any ideas? 
[liveboot.log](https://github.com/LSPosed/LSPosed/files/11931091/liveboot.log)
[magisk.log](https://github.com/LSPosed/LSPosed/files/11931092/magisk.log)
[magisk_log_2023-07-02T10.08.15.log](https://github.com/LSPosed/LSPosed/files/11931093/magisk_log_2023-07-02T10.08.15.log)
![WhatsApp Image 2023-07-02 at 10 08 04](https://github.com/LSPosed/LSPosed/assets/138396537/f7816339-441f-428e-8e44-c07d81b6455d)
![WhatsApp Image 2023-07-02 at 10 08 51](https://github.com/LSPosed/LSPosed/assets/138396537/9791d7e6-4cb3-4ba7-9242-7d7f22aaaf93)
![WhatsApp Image 2023-07-02 at 10 09 05](https://github.com/LSPosed/LSPosed/assets/138396537/2e72ba84-2ba1-48ce-a5a9-1f3abbb6b0f2)

**Reply by winerider** [reply]
Man ! I was having similar issue I just now installed this "LSPosed-v1.8.6-6950-zygisk-release.zip".
Try this it will work for those having similar kind of issues since the last update recently.

**Comment by ttlns** [comment]
Same issue, got the following log:
[verbose_2023-08-04T16:40:18.368803.log](https://github.com/LSPosed/LSPosed/files/12261909/verbose_2023-08-04T16.40.18.368803.log)
- Oneplus 11
- Android 13

**Reply by shellster** [reply]
Seems like LSPosed was properly broken since the last OTA for Android 13.  I'm also having this issue and so are a ton of people: https://forum.xda-developers.com/t/lsposed-module-says-not-installed-even-after-successful-installation.4609877/

For me it is manifesting a bit differently.  I already had LSPosed installed before but suddenly everything broke so I installed LSPosed again over top of the existing install.  It "activated successfully in Magisk", and when I got to LSPosed, the main page says it is active, however whenever I install an LSPosed module it does not show up in the module list in LSposed.  When I go to the "Hide My Applist" LSPosed module it says the "Module Not Activated".  It was before this OTA, and from all accounts it still is as apps that previously didn't work without HideMyApplist continue to work and not detect root apps.

**Reply by winerider** [reply]
Man ! I was having similar issue I just now installed this "LSPosed-v1.8.6-6950-zygisk-release.zip".
Try this it will work for those having similar kind of issues since the last update recently.

**Reply by ttlns** [reply]
Yep, that fixed it for me

**Reply by shellster** [reply]
@winerider Where do I find that release.  The latest one I see is: "LSPosed-v1.8.6-6712-zygisk-release.zip"

**Comment by ttlns** [comment]
@shellster 
at https://github.com/LSPosed/LSPosed/actions
you can find all releases

**Reply by shellster** [reply]
Thanks.  I tried the latest release and it solved the LSPosed not showing modules issue, but "HideMyAppList" still says the system service isn't running.  I'll try the exact version mentioned above when I get the chance.

**Reply by shellster** [reply]
Can confirm that "LSPosed-v1.8.6-6950-zygisk-release.zip" works and later releases (at least the latest version) do not.

**Comment by Devprince97** [comment]
I am still facing the issue

**Reply by shellster** [reply]
Did you upgrade/downgrade to the version of LSPosed mentioned in the previous comment?

**Comment by bzbzdxm** [comment]
I believe the problem lies in your opening of fingerprint authorization for the permissions.
